qiskit/circuit/quantumcircuit.py

```diff
@@ -2265,17 +2265,97 @@ def reset(self, qubit: QubitSpecifier) -> InstructionSet:
-        """Measure quantum bit into classical bit (tuples).
+        """Measure quantum bit (qubit) into classical bit (cbit).
 
-        Args:
-            qubit: qubit to measure.
-            cbit: classical bit to place the measurement in.
+        When a qubit is measured, its state collapses to a classical bit and copied to a
+        classical wire.
 
-        Returns:
-            qiskit.circuit.InstructionSet: handle to the added instructions.
+        In this example, a qubit is measured and the result of that measurement is stored in the
+        classical bit (usually expressed in diagrams as a double line):
 
-        Raises:
-            CircuitError: if arguments have bad format.
+        .. code-block::
+
+           from qiskit import QuantumCircuit
+           circuit = QuantumCircuit(1,1)
+           circuit.h(0)
+           circuit.measure(0, 0)
+           circuit.draw()
+
+
+        .. parsed-literal::
+
+                  ┌───┐┌─┐
+               q: ┤ H ├┤M├
+                  └───┘└╥┘
+             c: 1/══════╩═
+                        0
+
+        It is possible to call ``measure`` with lists of qubits and cbits of the same size:
+
+        .. code-block::
+
+           circuit = QuantumCircuit(2,2)
+           circuit.measure([0,1], [0,1])  # same as "circuit.measure(0,0); circuit.measure(1,1);"
+
+
+        .. parsed-literal::
+
+                  ┌─┐
+             q_0: ┤M├───
+                  └╥┘┌─┐
+             q_1: ─╫─┤M├
+                   ║ └╥┘
+             c: 2/═╩══╩═
+                   0  1
+
+        It is also possible to do a one-to-many readout:
+
+        .. code-block::
+
+           circuit = QuantumCircuit(2,2)
+           circuit.measure(0, [0,1])  # same as "circuit.measure(0,0); circuit.measure(0,1);"
+
+
+        .. parsed-literal::
+
+                  ┌─┐┌─┐
+             q_0: ┤M├┤M├
+                  └╥┘└╥┘
+             q_1: ─╫──╫─
+                   ║  ║
+             c: 2/═╩══╩═
+                   0  1
+
+        Instead of lists, you can use :class:`~qiskit.circuit.QuantumRegister` and
+        :class:`~qiskit.circuit.ClassicalRegister` under the same logic.
+
+        .. code-block::
+
+            from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister
+            qreg = QuantumRegister(1, "qreg")
+            creg = ClassicalRegister(2, "creg")
+            circuit = QuantumCircuit(qreg, creg)
+            circuit.measure(qreg, creg)
+
+
+        .. parsed-literal::
+
+                    ┌─┐┌─┐
+              qreg: ┤M├┤M├
+                    └╥┘└╥┘
+            creg: 2/═╩══╩═
+                     0  1
+
+
+         Args:
+             qubit: qubit/s to measure.
+             cbit: classical bit/s to place the measurement in.
+
+         Returns:
+             qiskit.circuit.InstructionSet: handle to the added instructions.
+
+         Raises:
+             CircuitError: if arguments have bad format.
```

Comment:
It's a bit odd to put these all the way down at the bottom.  They're pretty critical information, whereas all the extra examples above are supplementary.

The formatting problem is that these sections are indented by an extra space.

Reply:
Moving it under an _Example_ section, after Args/Return/Raise. It seems that there is an issue in the style, very much like https://github.com/Qiskit/qiskit_sphinx_theme/issues/45. I will report it on the theme.

<img width="455" src="https://user-images.githubusercontent.com/766693/222545715-caf6d437-bf8e-4027-ad0a-024c54043b05.png">